src/tools/compiletest/src/runtest.rs

```diff
@@ -309,10 +316,18 @@ impl<'test> TestCx<'test> {
         }
     }
 
+    fn effective_pass_mode(&self) -> Option<PassMode> {
```

Comment:
If it is possible to ensure through privacy that `props.pass_mode` is never used by itself, only through `effective_pass_mode`, I'd do that and then rename `effective_pass_mode` to just `pass_mode`.